core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -167,6 +167,16 @@ private TableProperties() {
       "write.delete.parquet.row-group-check-max-record-count";
   public static final int PARQUET_ROW_GROUP_CHECK_MAX_RECORD_COUNT_DEFAULT = 10000;
 
+  public static final String DEFAULT_PARQUET_BLOOM_FILTER_ENABLED = "write.parquet.bloom-filter-enabled.default";
```

Comment:
does it ever make sense to enable bloom filter for all columns? should we only allow bloom filter for explicitly specified columns?

Reply:
It is not a common usage to enable bloom filter for all columns, but it's legal. This is consistent with the parquet-mr bloom filter implementations.

Reply:
I agree with Steven. I think it only makes sense to enable bloom filters for some columns.

Reply:
+1. While it's consistent with the parquet-mr bloom filter implementaiton, we need to think of user experience first and foremost.

It doesn't make sense to enable bloom filters for _a lot_ of columns. And many users don't do any tuning of their metadata / statistics.

I think it's in-line with other things we do to make the users experience better, like turning off column level statistics after a certain number of columns. We can point it out in the docs under a big `!!!NOTE` (that's highlighted) that bloom filter is only used when turned on.

It's really an advanced thing to use at all imo.

Reply:
Sounds good. I have removed `DEFAULT_PARQUET_BLOOM_FILTER_ENABLED`. Now user needs to enable bloom filter for individual column using `PARQUET_BLOOM_FILTER_COLUMN_ENABLED_PREFIX`. If the column is a complex type, user needs to enable the column inside the complex type, for example, 
`set(PARQUET_BLOOM_FILTER_COLUMN_ENABLED_PREFIX + "struct_col.int_field", "true")`